controllers/snapshot/snapshot_adapter.go

```diff
@@ -248,16 +297,32 @@ TestScenarioLoop:
 		scenarioLabelAndKey := map[string]string{gitops.SnapshotTestScenarioLabel: integrationTestScenario.Name}
 		binding, err := a.createSnapshotEnvironmentBindingForSnapshot(a.application, copyEnv, a.snapshot, components, scenarioLabelAndKey)
 		if err != nil {
-			a.logger.Error(err, "Failed to create snapshotEnvironmentbinding for snapshot",
+			errStr := "Failed to create SnapshotEnvironmentBinding for snapshot"
+			a.logger.Error(err, errStr,
 				"snapshot", a.snapshot.Name,
 				"environment.Name", copyEnv.Name,
 				"snapshot.Spec.Components", a.snapshot.Spec.Components)
+
+			testStatuses.UpdateTestStatusIfChanged(
+				integrationTestScenario.Name, gitops.IntegrationTestStatusEnvironmentProvisionError,
+				fmt.Sprintf("%s: %s", errStr, err))
+			a.writeIntegrationTestStatusAtError(testStatuses)
+
 			return controller.RequeueWithError(err)
 		}
 		a.logger.LogAuditEvent("A snapshotEnvironmentbinding is created", binding, h.LogActionAdd,
 			"environment.Name", copyEnv.Name,
 			"integrationTestScenario.Name", integrationTestScenario.Name)
+
+		// TODO: I've no idea how to make this status update when conflict happen and reconciliation retriggers without interferring with other controllers
```

Comment:
:eyes: TODO

Race condition reproducer:
* have scenario A and B
* (snapshot) A is marked as in progress by snapshot adapter (not written yet into annotation)
* (integrationpipeline) status of A is updated by integration and written into snapshot
* (snapshot) B is marked as in progress
* (snapshot) tries to write annotation into snapshot, but fails on conflict becasuse it was updated by integrationpipeline controller, requeue with error
* At this point, because SEB for test B already exist, code branch for setting status to inProgress is not executed and thus B is still marked as inProgress
* Eventually, when PLR B starts status will be updated and sort of self fixed

One potential fix here, is to add another condition when SEB already exists set to InProgress if current status is Pending.
WDYT, is it worth fixing or overengineerd?

Reply:
sounds like simple fix, I probably will go with that if you agree

Reply:
For me it sounds like worth fixing. Its possible that in future there will be simmilar element to SEB that will be running into same exact issue, we could apply fixed approach to it also.

Reply:
updated

Reply:
If it may have some confilcts  that updating the status to snapshot, do we consider to introduce some lock mechanism. Such as Mutex？

Reply:
That may bring even more complications by handling this in advance, k8s reconciliation is based on optimistic merging, given and I'm not sure if we can even do it, as it's updated by  isolated parallel processes. However I resolved this particular issue already.

Reply:
OK. I just thought that it will need a lot of locks.